src/main/java/com/vackosar/gitflowincrementalbuild/boundary/UnchangedProjectsRemover.java

```diff
@@ -1,50 +1,43 @@
 package com.vackosar.gitflowincrementalbuild.boundary;
 
-import com.google.inject.Singleton;
 import com.vackosar.gitflowincrementalbuild.control.ChangedProjects;
 import org.apache.maven.execution.MavenSession;
 import org.apache.maven.project.MavenProject;
 import org.codehaus.plexus.logging.Logger;
 import org.eclipse.jgit.api.errors.GitAPIException;
 
 import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
 import java.io.IOException;
 import java.util.*;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 @Singleton
+@Named("gib.unchangedProjectsRemover")
 class UnchangedProjectsRemover {
 
     private static final String MAVEN_TEST_SKIP = "maven.test.skip";
     private static final String MAVEN_TEST_SKIP_EXEC = "skipTests";
     static final String TEST_JAR_DETECTED = "Dependency with test-jar goal detected. Will compile test sources.";
     private static final String GOAL_TEST_JAR = "test-jar";
 
-    private Configuration configuration;
-    private Logger logger;
-    private ChangedProjects changedProjects;
-    private MavenSession mavenSession;
-
-    @Inject
-    UnchangedProjectsRemover(Configuration configuration, Logger logger, ChangedProjects changedProjects,
-            MavenSession mavenSession) {
-        this.configuration = configuration;
-        this.logger = logger;
-        this.changedProjects = changedProjects;
-        this.mavenSession = mavenSession;
-    }
+    @Inject private Logger logger;
+    @Inject private ChangedProjects changedProjects;
+    @Inject private MavenSession mavenSession;
```

Comment:
any chance of injecting into static fields?

Reply:
I don't think that'll work. But what would be the benefit here anyway?